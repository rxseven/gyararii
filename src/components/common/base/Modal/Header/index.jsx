import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

const propTypes = exact({
  children: PropTypes.node.isRequired,
  onClose: PropTypes.func.isRequired
});

class Header extends React.Component {
  handleClose = () => {
    const { onClose } = this.props;

    onClose();
  };

  render() {
    const { children } = this.props;

    return (
      <div className="modal-header">
        <h5 className="modal-title">{children}</h5>
        <button
          aria-label="Close"
          className="close"
          onClick={this.handleClose}
          type="button"
        >
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
    );
  }
}

Header.propTypes = propTypes;

export default Header;
