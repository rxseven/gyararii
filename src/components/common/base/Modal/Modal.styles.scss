// Bootstrap Modal
// https://getbootstrap.com/docs/4.1/components/modal/

.modal-open {
  .modal {
    &,
    &.fade {
      display: block;
    }
  }
}

.modal {
  background-color: rgba(0, 0, 0, 0.8);
}
