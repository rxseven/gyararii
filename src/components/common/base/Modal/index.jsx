import PropTypes from 'prop-types';
import React from 'react';
import ReactModal from 'react-modal';

import './Modal.styles.scss';

// Hide the application from screenreaders and other assistive technologies
if (process.env.NODE_ENV !== 'test') ReactModal.setAppElement('#root');

const propTypes = {
  children: PropTypes.node.isRequired,
  isOpen: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired
};

function Modal({ children, onClose, ...props }) {
  return (
    <ReactModal
      {...props}
      bodyOpenClassName="modal-open"
      className="modal-dialog"
      onRequestClose={onClose}
      overlayClassName="modal"
    >
      <div className="modal-content">{children}</div>
    </ReactModal>
  );
}

Modal.propTypes = propTypes;

export default Modal;
