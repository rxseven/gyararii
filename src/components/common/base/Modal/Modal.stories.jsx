import React from 'react';
import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';

import mock from 'stories/mock';
import Modal from './index';

const { text } = mock;
const title = 'Components/Common/Base/Modal';
const props = { isOpen: false, onClose: action('onClose') };

storiesOf(title, module)
  .add('close (default)', () => <Modal {...props}>{text}</Modal>, {
    notes: 'A component is invisible by default (isOpen: false).'
  })
  .add('open', () => (
    <Modal {...props} isOpen>
      {text}
    </Modal>
  ))
  .add('assembled', () => (
    <Modal {...props} isOpen>
      <Modal.Header onClose={action('onClose')}>Header</Modal.Header>
      <Modal.Body>{text}</Modal.Body>
      <Modal.Footer>Footer</Modal.Footer>
    </Modal>
  ));
