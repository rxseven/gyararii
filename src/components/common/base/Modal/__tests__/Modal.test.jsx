import { shallow } from 'enzyme';
import React from 'react';

import Modal from '../index';

describe('<Modal />', () => {
  // Arrange
  const props = { children: 'content', isOpen: true, onClose: jest.fn() };
  const component = <Modal {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });
});
