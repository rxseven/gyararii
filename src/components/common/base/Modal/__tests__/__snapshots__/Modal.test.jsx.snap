// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Snapshot tests should render correctly 1`] = `null`;
